app/gui/view/graph-editor/src/free_place_finder.rs

```diff
@@ -0,0 +1,255 @@
+//! The module containing an algorithm for searching free space for some point going at the specific
+//! direction.
+//!
+//! This is used in Graph Editor to find unoccupied place for newly created node.
+
+use crate::prelude::*;
+
+use ordered_float::OrderedFloat;
+
+
+
+// ====================
+// === OccupiedArea ===
+// ===================
+
+/// The structure describing an occupied area.
+///
+/// All such areas are rectangles described by x and y ranges.
```

Comment:
`x1` is start of the range and `x2` is end of it? If so, add short info about it pls.

EDIT: after seeing implementation, it seems that `x1` and `x2` are not sorted. This needs short doc :)